### PR TITLE
Update README.md for monadPattern

### LINKED ISSUE #2245: Explanation for Monad
In this issue, let's add a proper explanation for the pattern.

Acceptance criteria

- The pattern's README.md has been amended with an explanation
- The explanation consists of
  - Real world example
  - In plain words
  - Wikipedia says
  - Programmatic example

### PR DESCRIPTION
**Pull request by Chant3ll3**
Fixes #2245 Explanation for Monad

Updating Monad pattern Readme.md file.

### CODE REVIEW

**Review by susanhsrestha — APPROVED** [review]

File: monad/README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ title: Monad
 category: Functional
 language: en
 tag:
- - Reactive
+- Reactive
```
Comment (devas10) [inline monad/README.md:6]: Avoid removing spaces. They are meant for indentation

Comment (Chant3ll3) [reply monad/README.md]: I'm sorry, I have added the space back

**Review by devas10 — APPROVED** [review]
seems good

**Review by susanhsrestha — APPROVED** [review]

**Review by iluwatar — APPROVED** [review]

**Comment by iluwatar** [comment]
Looks good! Thank you for the contribution 🎉

@all-contributors please add @Chant3ll3 for docs

**Pull request merged**